### PR TITLE
EZP-27314: Add fetch and Element.closest polyfills to support Safari 10 and Edge 14

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-27314

# Description

`ez-platform-ui-app` relies on relatively new API that are not yet supported in every *supported* browser. At the moment, we have the following issues:

* Safari 10.0 does not support `fetch`
* Edge 14 lacks `Element.closest`

This PR adds the corresponding polyfills to dev dependencies so that unit tests can fully run with those browsers. Following a custom element/web components/polymer *good* practice, `ez-platform-ui-app` does not include those polyfills when used in a web page. Instead, `fetch` and `Element.closest` are documented as required API and that Safari 10.0 and Edge 14 will need the corresponding polyfill so that this element works as expected.

This PR also fixes a test logic issue that ironically was only failing in Edge and Safari and fixes a failing test in Edge due its non support for `addEventListener` `options` parameter.

### CODE REVIEW

**Comment by andrerom** [comment]
Is there some way we can avoid loading polyfils if browsers don't need it? Asking as I guess the list will grow.. ;)

**Comment by dpobel** [comment]
> Is there some way we can avoid loading polyfils if browsers don't need it?

that's a very good question @andrerom. I got the same concern and I was digging around that topic lately and this morning I stumbled upon several videos mentioning the polyfill topic. Basically, it seems the good practice is to not include any polyfill in a custom element but to document its requirement and let the consumer of the custom element handle the polyfill (so exactly the opposite of what I did :)). So the current patch is probably not the good solution but since we are both producer and consumer of `ez-platform-ui-app`, we still need to figure out how we'll load those polyfills but later ;-)

> Asking as I guess the list will grow.. ;)

I'm not that sure. Our supported browser baseline is quite advanced. Also for those 2 polyfills, we have 2 different cases:

* for `fetch`: we need the polyfill only because of Safari 10.0. I'm unsure whether this requirement will remain until 2.0 is out :) And if it stays we could also just switch to a raw XHR, that would not be a big task (maybe easier than finding an efficient way of loading the polyfill ;))
* for `closest`: we need it for Edge 14 so the same browser support question applies also for it. But [the polyfill is really super small](https://github.com/jonathantneal/closest/blob/master/element-closest.js) (830 bytes **without** minification and gzip), we could perfectly add it inline in the page running app, it would be almost invisible :)

So to summarize, I'm about to change this PR to include the polyfills only in the test for now and to document those requirements. We'll see later how we handle those.

**Comment by dpobel** [comment]
actually tests are still failing in Edge 14 (time out), I'm on it...
